src/Shared/CommunicationsUtilities.cs

```diff
@@ -355,64 +355,86 @@ internal static void WriteLongForHandshake(this PipeStream stream, long value)
         /// <summary>
         /// Extension method to read a series of bytes from a stream
         /// </summary>
-        internal static long ReadLongForHandshake(this PipeStream stream)
+        internal static long ReadLongForHandshake(this PipeStream stream
+#if NETCOREAPP2_1
+            , int handshakeReadTimeout
```

Comment:
What if instead of ifdefs you leave the timeout in and pass 0 on non-netcore?

Reply:
The intention for the ifdefs is to avoid touching the output of the full framework build since we don't _have_ to and it's in escrow. It's definitely not nice clean code :(